utils/docker_utils.go

```diff
@@ -0,0 +1,277 @@
+//cli - the docker client
+//id - the container id
+//It returns:
+//1. an error if the passed id doesn't exists
+//2. nil otherwise.
+func RemoveContainer(cli *client.Client, id string) error {
+	log.Printf("Removing Container [%s]", id)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	return cli.ContainerRemove(ctx, id, types.ContainerRemoveOptions{})
+}
+
+//Iterates over the content and write each one to the destination file inside the container
+//Params:
+//cli - the docker client
+//id - the container id
+//content - the array that stores the content.
+//Each position of the array will become a line in the dest file
+//dest - the destination file path, inside the container.
+//It returns:
+//1. an error if the passed id doesn't exists or if the destination file is a invalid one
+//2. nil otherwise.
+func Write(cli *client.Client, id string, content []string, dest string) error {
+	for _, c := range content {
+		c = strings.ReplaceAll(c, "'", "'\"'\"'")
+		cmd := fmt.Sprintf(`echo -E '%s' >> %s`, c, dest)
+		log.Printf("Writing [%s] on [%s] from Container [%s]", c, dest, id)
+		err := Exec(cli, id, cmd)
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//It copies the src file, which lives in the worker host,
+//to the dest file inside the container.
+//Params:
+//cli - the docker client
+//id - the container id
+//src - the source file path (in the worker host)
+//dest - the destination file, inside the container
+//It returns:
+//1. an error if the passed id doesn't exists or if the destination file is a invalid one
+//2. nil otherwise.
+func Copy(cli *client.Client, id, src, dest string) error {
+	log.Printf("Copy [%s] to [%s] from Container [%s]", src, dest, id)
+	dat, _ := ioutil.ReadFile(src)
+	content := string(dat)
+	content = strings.ReplaceAll(content, "'", "'\"'\"'")
+	cmd := fmt.Sprintf("echo -E '%s' >| %s", content, dest)
+	return Exec(cli, id, cmd)
+}
+
+//Executes a bash command inside the container
+//Params:
+//cli - the docker client
+//id - the container id
+//cmd - the bash command (e.g "echo 'arrebol'")
+//It returns:
+//1. an error if the command couldn't be executed inside the container
+//(e.g call a binary that doesn't exists), or if the id doesn't exists
+//2. nil otherwise.
+func Exec(cli *client.Client, id, cmd string) error {
+	log.Printf("Executing command [%s] on container [%s]", cmd, id)
+	config := types.ExecConfig{
+		Cmd: []string{"/bin/bash", "-c", cmd},
+	}
+	rid, _ := cli.ContainerExecCreate(context.Background(), id, config)
+	return cli.ContainerExecStart(context.Background(), rid.ID, types.ExecStartCheck{})
+}
+
+//Executes cat in a file inside the container and returns its output
+//Params:
+//cli - the docker client
+//id - the container id
+//path - the file path inside the container
+//It returns:
+//1. nil and an error if the id doesn't exists,
+//or if the file path is invalid.
+//2. The file content as byte array and nil otherwise.
```

Comment:
rename this function. Maybe, to `read(cli, id, path) ([]byte, error)`

Reply:
this function actually reads the content that comes from a connection (a socket). See Cat func, it uses the aux read func to read the content of a hijack connection. In short, the cat executes the unix `cat` inside the container and expose the result to the hijack connection, then, the read func get this content.

Reply:
still not convinced. rename to `read` as I suggested. `cat` means `concat`.

Reply:
done